pkg/protocol/handshake/message_server_key_exchange.go

```diff
@@ -65,7 +68,15 @@ func (m *MessageServerKeyExchange) Unmarshal(data []byte) error {
 	if _, ok := elliptic.CurveTypes()[elliptic.CurveType(data[0])]; ok {
 		m.EllipticCurveType = elliptic.CurveType(data[0])
 	} else {
-		return errInvalidEllipticCurveType
+		// checking for ECDHE_PSK_*
```

Comment:
Parsing with heuristics seems dangerous to me, does the RFC recommend a different way?

I worry that parse fail + move pointer try again could lead to a bug.

Reply:
introduced `KeyExchangeAlgorithm`  as is in [RFC4279](https://datatracker.ietf.org/doc/html/rfc4279)